crates/trigger/src/runtime_config/variable_provider.rs

```diff
@@ -1,28 +1,30 @@
 use std::path::PathBuf;
 
 use serde::Deserialize;
-use spin_config::provider::{env::EnvProvider, vault::VaultProvider};
+use spin_variables::provider::{env::EnvProvider, vault::VaultProvider};
 
 use super::RuntimeConfig;
 
-pub type ConfigProvider = Box<dyn spin_config::Provider>;
+pub type VariableProvider = Box<dyn spin_variables::Provider>;
 
-const DEFAULT_ENV_PREFIX: &str = "SPIN_CONFIG";
+const DEFAULT_ENV_PREFIX: &str = "SPIN_VARIABLE";
```

Comment:
I think we should first try `SPIN_VARIABLE_`, then try `SPIN_CONFIG_` and _if found, print a deprecation notice_.

Reply:
@itowlson added a deprecation check, let me know your thoughts on the language!

Reply:
Sample output:

```
❯ SPIN_CONFIG_OBJECT="teapot" spin up
Logging component stdio to ".spin/logs/"

Serving http://127.0.0.1:3000
Available Routes:
  spin-variales-rust: http://127.0.0.1:3000 (wildcard)
Use of environment provided variables prefix SPIN_CONFIG is deprecated
```

Reply:
I wonder if it would be okay to be a bit more specific than that e.g.

> Warning: variable 'foo': SPIN_VARIABLE_FOO was not set, so used SPIN_CONFIG_FOO. The SPIN_CONFIG prefix is deprecated; please switch to the SPIN_VARIABLE prefix.

At some point we should write a docs page on v1-to-v2 changes, and when we do that we can link it in the warning.

(I do worry it might be a bit wordy if it was printed repeatedly!  But we can see how that feels in practice and get clever if it proves bothersome.)